ding/utils/data/dataset.py

```diff
@@ -50,8 +58,13 @@ def __init__(self, cfg: dict) -> None:
             d4rl.set_dataset_path(data_path)
         env = gym.make(env_id)
         dataset = d4rl.qlearning_dataset(env)
-        if cfg.policy.collect.get('normalize_states', None):
-            dataset = self._normalize_states(dataset)
+        self._cal_statistics(dataset, env)
+        try:
+            if cfg.env.norm_obs.use_norm and cfg.env.norm_obs.offline_stats.use_offline_stats:
```

Comment:
maybe we can reduce nested layer to just one, e.g., `cfg.env.use_norm` and `cfg.env.use_offline_stats`

Reply:
You are right. But 
1. I use `cfg.env.norm_obs.use_norm` for backward compatibility, otherwise, I need to change a lot of configs.
2. Semantically, `use_offline_stats` controls how `norm_obs` is conducted. So it makes more sense it have it in the `norm_obs` dict although it leads to more codes.
3. `offline_stats` not only includes boolean `use_offline_stats` but also includes the `mean` and `std` calculated from the offline dataset, so I think we should keep `offline_stats` dict in addition to `use_offline_stats` flag.

Reply:
ok, we will replace name in the future